newm-server/src/main/kotlin/io/newm/server/features/scheduler/ReferralHeroSyncJob.kt

```diff
@@ -0,0 +1,43 @@
+package io.newm.server.features.scheduler
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.newm.server.features.referralhero.repo.ReferralHeroRepository
+import io.newm.server.features.user.database.UserEntity
+import io.newm.shared.koin.inject
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.quartz.DisallowConcurrentExecution
+import org.quartz.Job
+import org.quartz.JobExecutionContext
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+
+@DisallowConcurrentExecution
+class ReferralHeroSyncJob : Job {
+    private val log = KotlinLogging.logger {}
+    private val referralHeroRepository: ReferralHeroRepository by inject()
+
+    override fun execute(context: JobExecutionContext) {
+        log.info {
+            "ReferralHeroSyncJob key: ${context.jobDetail.key.name} executed at ${
+                LocalDateTime.ofInstant(context.fireTime.toInstant(), ZoneOffset.UTC)
+            }"
+        }
+
+        val users = transaction { UserEntity.allWithoutReferralCode().toList() }
+        log.info { "Found ${users.size} users to sync" }
+        for (user in users) {
+            try {
```

Comment:
Move try catch to capture everything in the execute function. If we throw an exception (db connection issue), then the job will not try again in 24 hours.

Reply:
Good callout, done!